File: app/src/main/java/mb00/android/codehub/ui/search/view/SearchActivity.java
package mb00.android.codehub.ui.search.view;

import android.os.Bundle;
import android.support.v4.view.PagerAdapter;

import mb00.android.codehub.R;
import mb00.android.codehub.data.BundleKeys;
import mb00.android.codehub.databinding.ActivitySearchBinding;
import mb00.android.codehub.ui.base.view.BaseDrawerActivity;
import mb00.android.codehub.ui.home.view.HomeActivity;
import mb00.android.codehub.ui.search.adapter.SearchFragmentPagerAdapter;
import mb00.android.codehub.ui.search.viewmodel.SearchViewModel;

/**
 * Launched from {@link HomeActivity} if searchButton is clicked
 * Immediately launches {@link SearchFragmentPagerAdapter}
 */

public class SearchActivity extends BaseDrawerActivity<ActivitySearchBinding, SearchViewModel> {

    private Bundle queryBundle;
    private String searchQuery;

    @Override
    protected int layout() {
        return R.layout.activity_search;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initSearchQuery();
        initSearchListeners();
        initViewPager();
    }

    private void initSearchQuery() {
        queryBundle = new Bundle();
        searchQuery = "";
        queryBundle.putString(BundleKeys.SEARCH_QUERY_KEY, searchQuery);
    }

    private void initSearchListeners() {
        getBinding().searchQueryEditText.setOnEditorActionListener(((textView, i, keyEvent) -> {
            search();
            return true;
        }));
    }

    private void initViewPager() {
        PagerAdapter searchPagerAdapter = new SearchFragmentPagerAdapter(getSupportFragmentManager(), this, queryBundle);
        getBinding().searchViewPager.setAdapter(searchPagerAdapter);
        getBinding().searchViewPager.setOffscreenPageLimit(3);
        getBinding().searchViewPager.setPageMargin(4);
        getBinding().searchViewPager.setPageMarginDrawable(R.color.grey);
        getBinding().searchTabLayout.setupWithViewPager(getBinding().searchViewPager);
    }

    public void search() {
        searchQuery = "";
        queryBundle.putString(BundleKeys.SEARCH_QUERY_KEY, searchQuery);
        searchQuery = getBinding().searchQueryEditText.getText().toString();
        queryBundle.putString(BundleKeys.SEARCH_QUERY_KEY, searchQuery);
        getBinding().searchViewPager.getAdapter().notifyDataSetChanged();
    }

}

File: app/src/main/java/mb00/android/codehub/ui/search/view/SearchActivity.kt
package mb00.android.codehub.ui.search.view

import android.os.Bundle
import mb00.android.codehub.R
import mb00.android.codehub.data.BundleKeys
import mb00.android.codehub.databinding.ActivitySearchBinding
import mb00.android.codehub.ui.base.view.BaseDrawerActivity
import mb00.android.codehub.ui.home.view.HomeActivity
import mb00.android.codehub.ui.search.adapter.SearchFragmentPagerAdapter
import mb00.android.codehub.ui.search.viewmodel.SearchViewModel

/**
 * Launched from [HomeActivity] if searchButton is clicked
 * Immediately launches [SearchFragmentPagerAdapter]
 */

class SearchActivity : BaseDrawerActivity<ActivitySearchBinding, SearchViewModel>() {

    private var queryBundle: Bundle? = null
    private var searchQuery: String? = null

    override fun layout(): Int {
        return R.layout.activity_search
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initSearchQuery()
        initSearchListeners()
        initViewPager()
    }

    private fun initSearchQuery() {
        queryBundle = Bundle()
        searchQuery = ""
        queryBundle?.putString(BundleKeys.SEARCH_QUERY_KEY, searchQuery)
    }

    private fun initSearchListeners() {
        binding.searchQueryEditText.setOnEditorActionListener { _, _, _ ->
            search()
            true
        }
    }

    private fun initViewPager() {
        val searchPagerAdapter = SearchFragmentPagerAdapter(supportFragmentManager, this, queryBundle!!)
        binding.searchViewPager.adapter = searchPagerAdapter
        binding.searchViewPager.offscreenPageLimit = 3
        binding.searchViewPager.pageMargin = 4
        binding.searchViewPager.setPageMarginDrawable(R.color.grey)
        binding.searchTabLayout.setupWithViewPager(binding.searchViewPager)
    }

    fun search() {
        searchQuery = ""
        queryBundle?.putString(BundleKeys.SEARCH_QUERY_KEY, searchQuery)
        searchQuery = binding.searchQueryEditText.text.toString()
        queryBundle?.putString(BundleKeys.SEARCH_QUERY_KEY, searchQuery)
        binding.searchViewPager.adapter?.notifyDataSetChanged()
    }

}
